sympy/matrices/determinant.py

```diff
@@ -327,19 +329,29 @@ def _adjugate(M, method="berkowitz"):
     return M.cofactor_matrix(method=method).transpose()
 
 
+def _charpoly_DOM(M, x):
+    DOM = DomainMatrix.from_Matrix(M, field=True, extension=True)
+    x = uniquely_named_symbol(x, M, modify=lambda s: '_' + s)
+    K = DOM.domain
+    coeff = DOM.charpoly()
+    coeffs = coeff
+    if K == EX:
+        coeffs = []
+        for e in coeff:
+            coeffs.append(K.from_sympy(e))
```

Comment:
Is it necessary to do this? Can the EX domain elements not be passed directly to Poly?

Reply:
Sure, it will work without this condition, my bad.
Thanks.